qdax/tasks/jumanji_envs.py

```diff
@@ -0,0 +1,189 @@
+
+        action = jnp.argmax(proba_action)
+
+        state_desc = None
+        next_state, next_timestep = env.step(env_state, action)
+
+        next_state_desc = None
+
+        transition = QDTransition(
+            obs=timestep.observation,
+            next_obs=next_timestep.observation,
+            rewards=next_timestep.reward,
+            dones=jnp.where(next_timestep.last(), x=jnp.array(1), y=jnp.array(0)),
+            actions=action,
+            truncations=jnp.array(0),
+            state_desc=state_desc,
+            next_state_desc=next_state_desc,
+        )
+
+        return next_state, next_timestep, policy_params, random_key, transition
+
+    return default_play_step_fn
+
+
+@partial(jax.jit, static_argnames=("play_step_fn", "episode_length"))
+def generate_jumanji_unroll(
+    init_state: JumanjiState,
+    init_timestep: JumanjiTimeStep,
+    policy_params: Params,
+    random_key: RNGKey,
+    episode_length: int,
+    play_step_fn: Callable[
+        [JumanjiState, JumanjiTimeStep, Params, RNGKey],
+        Tuple[
+            JumanjiState,
+            JumanjiTimeStep,
+            Params,
+            RNGKey,
+            Transition,
+        ],
+    ],
+) -> Tuple[JumanjiState, JumanjiTimeStep, Transition]:
+    """Generates an episode according to the agent's policy, returns the final state of
+    the episode and the transitions of the episode.
+
+    Args:
+        init_state: first state of the rollout.
+        policy_params: params of the individual.
+        random_key: random key for stochasiticity handling.
+        episode_length: length of the rollout.
+        play_step_fn: function describing how a step need to be taken.
+
+    Returns:
+        A new state, the experienced transition.
+    """
+
+    def _scan_play_step_fn(
+        carry: Tuple[JumanjiState, JumanjiTimeStep, Params, RNGKey], unused_arg: Any
+    ) -> Tuple[Tuple[JumanjiState, JumanjiTimeStep, Params, RNGKey], Transition]:
+        env_state, timestep, policy_params, random_key, transitions = play_step_fn(
+            *carry
+        )
+        return (env_state, timestep, policy_params, random_key), transitions
+
+    (state, timestep, _, _), transitions = jax.lax.scan(
+        _scan_play_step_fn,
+        (init_state, init_timestep, policy_params, random_key),
+        (),
+        length=episode_length,
+    )
+    return state, timestep, transitions
+
+
+@partial(
+    jax.jit,
+    static_argnames=(
+        "episode_length",
+        "play_step_fn",
+        "behavior_descriptor_extractor",
+    ),
+)
+def jumanji_scoring_function(
+    policies_params: Genotype,
+    random_key: RNGKey,
+    init_states: JumanjiState,
+    init_timesteps: JumanjiTimeStep,
+    episode_length: int,
+    play_step_fn: Callable[
+        [JumanjiState, JumanjiTimeStep, Params, RNGKey, jumanji.env.Environment],
+        Tuple[JumanjiState, JumanjiTimeStep, Params, RNGKey, QDTransition],
+    ],
```

Comment:
Out of curiosity, why did you choose to make `play_step_fn` a mandatory argument? (contrary to QDax, where if `play_step_fn` is `None`, then it becomes automatically equal to the result of `make_policy_network_play_step_fn`)

Reply:
Actually, here it can't be. In the Brax part, it's optional in a make function but not in the scoring_function. And I did not add a make_fn for this one because I wanted to keep the PR minimal and add only the important elements.

Reply:
Ok! Sounds good! :)